canonical/maas-ui
# fix(auth): Log out user if session has expired MAASENG-1414

## Done

- Added `sessionExpired` to status state
- Set authenticated to false if session expires
- Add info notification on login screen if session expires
- Add warning message on session timeout page to inform users that session will expire if timeout length changed

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to `/settings/security/session-timeout`
- Change the timeout to "5 minutes" and save
- Ensure you are logged out (may have to wait a few seconds)
- Log in
- Make yourself a coffee/tea/beverage of your choice
- Chill out until 5 minutes have passed
- Go back to your computer
- Ensure you are logged out

## Fixes

- Fixes https://warthogs.atlassian.net/browse/MAASENG-1414
- Fixes #864 

## Screenshots

![image](https://user-images.githubusercontent.com/35104482/225609256-84e3c8b6-bee4-47fb-9170-73ed932e99a1.png)

![image](https://user-images.githubusercontent.com/35104482/225609300-db26850c-bc41-4824-8088-60532c374b54.png)
